shared/src/main/scala/StringHelpers.scala

```diff
@@ -15,6 +16,75 @@ object StringHelpers:
       case a: VNum  => a.toInt.toChar.toString
       case a: VList => VList(a.map(chrord)*)
 
+  // https://codegolf.stackexchange.com/a/151721/78850
+  def compressDictionary(s: String)(using ctx: Context): String =
+    val endLength = 2 + ctx.globals.longDictionary.map(_.length).max
+
+    def character(z: Int, c: Char) =
+      val o =
+        if c.toInt == 10 then 95
+        else if ' ' <= c && c <= '~' then c.toInt - 32
+        else throw new Exception(s"Invalid character $c")
+
+      3 * 96 * z + o
+
+    def dictionary(z: Int, w: String, nonempty: Boolean) =
+      var ts = nonempty
+      var subW = w
+      if w.headOption.exists(_ == ' ') then
+        subW = w.substring(1)
+        ts = !ts
+      val dictionary =
+        if w.size < 6 then ctx.globals.shortDictionary
+        else ctx.globals.longDictionary
+      val swapcase = !dictionary.exists(_ == w)
+      val word =
+        if !swapcase then w
+        else
+          val first = w.charAt(0)
+          val rest = w.substring(1)
+          val firstToggled =
+            if first.isUpper then first.toLower else first.toUpper
+          firstToggled.toString + rest
+
+      if !dictionary.contains(word) then
+        throw new Exception(s"Invalid word $word")
+      val f = ts || swapcase
+      val j = if swapcase then 2 else 1
+      val i = dictionary.indexOf(word)
+
+      var res = dictionary.size * z + i
+      if f then
+        res = 3 * res + j
+        res = 3 * res + 2
+      else res = 3 * res + 1
+      res
+    end dictionary
+
+    def go(z: Int) =
+      val compressed = StringBuilder()
+      var z1 = z
+      while z1 != 0 do
+        val c = z1 % 250
+        z1 = z1 / 250
+        compressed.append(CODEPAGE(c))
+      compressed.toString.reverse
+
+    val dp = Array.fill(s.length + 1)(0)
+    // scala equivalent of for i in range(len(str) -1,-1,-1)
+    for i <- (s.length - 1) to 0 by -1 do
+      dp(i) = character(dp(i + 1), s(i))
+      for j <- 1 to Math.min(endLength, s.length - i) do
+        try
+          dp(i) = Math.min(
+            dp(i),
+            dictionary(dp(i + j), s.substring(i, i + j), i != 0)
+          )
+        catch case _: Exception => () // todo (lyxal): is this necessary?
```

Comment:
@ysthakur yes it is - without it, words not in the dictionary cause the function to execute with an error. I assume that the errors thrown are for control execution.

Reply:
Then we should check if a word is in the dictionary before accessing it instead of catching all `Exception`s. Silently catching a whole category of exceptions will make debugging quite a bit harder